sample_structs_test.go

```diff
@@ -100,11 +97,33 @@ type InnerStruct struct {
 	StringField2     string `csv:"stringField2"`
 }
 
-var _ TypeUnmarshalCSVWithFields = (*UnmarshalCSVWithFieldsSample)(nil)
-
 type UnmarshalCSVWithFieldsSample struct {
 	Foo  string  `csv:"foo"`
 	Bar  int     `csv:"bar"`
 	Baz  string  `csv:"baz"`
 	Frop float64 `csv:"frop"`
 }
+
+func (u *UnmarshalCSVWithFieldsSample) UnmarshalCSVWithFields(key, value string) error {
+	switch key {
+	case "foo":
+		u.Foo = value
+	case "bar":
+		i, err := strconv.Atoi(value)
```

Comment:
strconv.Parse* is better than Atoi for a lot of good reasons, I'll leave the docs to explain this and assume if you're using it knowing that you have a good reason. just wanted to note it.